src/ui.rs

```diff
@@ -73,6 +75,50 @@ fn use_color(choice: ColorChoice) -> bool {
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum PaginationChoice {
+    Always,
+    Never,
+}
+
+impl Default for PaginationChoice {
+    fn default() -> Self {
+        PaginationChoice::Never
+    }
+}
+
+impl FromStr for PaginationChoice {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "always" => Ok(PaginationChoice::Always),
+            "never" => Ok(PaginationChoice::Never),
+            _ => Err("must be either always or never"),
+        }
+    }
+}
+
+fn use_pagination(choice: PaginationChoice) -> bool {
+    match choice {
+        PaginationChoice::Always => true,
+        PaginationChoice::Never => false,
+    }
+}
+
+fn pagination_setting(settings: &UserSettings) -> PaginationChoice {
+    settings
+        .config()
+        .get_string("ui.paginate")
+        .ok()
+        .and_then(|s| s.parse().ok())
+        .unwrap_or_default()
+}
+
+fn pager_setting(settings: &UserSettings) -> String {
+    settings.config().get_string("ui.pager").unwrap_or_default()
+}
```

Comment:
Hm, if we move this + the color choice stuff to `UserSettings`, then instead of doing `color_setting(settings)`, we'd have `settings.color`. That looks like it makes more sense, yeah.

Reply:
Yeah this looks better IMO

Reply:
We need to figure out at some point where things like this should go. It doesn't seem like it belongs in the lib crate (where `UserSettings` lives) since it's specific to the CLI. There were already a few similar cases, so I don't mean that we shouldn't add this config there; I'm just saying that we should think more about it at some point.

Reply:
Hmm, but this PR moves quite a lot of code that's obviously related to the CLI into the lib crate. The previous offenders just named a config variable and maybe a default value. It seems unfortunate to add those CLI-specific types to the lib crate if we know we're going to move them out later. What do you think?

Reply:
I also considered that.  I moved them into the `UserSettings` because `use_progress_indicator` is there, and that seems obviously CLI-based too. As long as all of these settings are in the same place, I don't feel _too_ strongly about it. Maybe I have a slight preference for `ui.rs`; it seems a bit more accurate and preserves the status quo.

Reply:
Right, there are already a few options that are obviously CLI-related. I also considered moving them to `ui.rs` (onto `Ui`, probably). I think that would be better.

Reply:
Okay. To avoid a lot of churn, I'll move them to `ui.rs`, but I won't add the fields to `Ui` (since there's a good chance we'll just end up moving them again).